src/transformers/models/wav2vec2_conformer/convert_wav2vec2_conformer_original_pytorch_checkpoint_to_pytorch.py

```diff
@@ -217,13 +220,21 @@ def load_conv_layer(full_name, value, feature_extractor, unused_weights, use_gro
         unused_weights.append(full_name)
 
 
+def import_user_module(user_dir: str):
+    arg = namedtuple("Arg", ["user_dir"])
+    arg = arg(user_dir.__str__())
+    utils.import_user_module(arg)
```

Comment:
What is this used for ? For my own curiosity

Reply:
@ylacombe 
This option is used when overriding task, model, and data with extensions in fairseq. [**[link]**](https://github.com/facebookresearch/fairseq/blob/main/fairseq/dataclass/configs.py#L208-L213)

Users who use custom overriding must enter the overriding extension path when loading the model. However, the default is to operate the same as the existing huggingface implementation.

Reply:
I'm not sure I fully understand why we need this - it's quite magic code for important extra modules. Are these modules actually required when converting the state dict? I'd assume that they're only used to change the task/data/model args, but not the weights themselves? Unless there's a clear use case for this, I would be in favour of simplifying the logic and removing the function

Reply:
@sanchit-gandhi  

When training with fairseq, we use the `user_dir` option when training by adding models, tasks, and modules as extensions.
So when we load a faiseq model, an extension code is required to load the model.
I think it's okay to disable it when `extension_path` is None, or just delete it because it's a special case.

[[link]](https://github.com/facebookresearch/fairseq/blob/main/examples/wav2vec/config/finetuning/vox_1h_aws.yaml#L3-L7)

Reply:
Sure! But does the user actually need to pass `user_dir` when **converting weights** (not training)? Surely all we need to do is be able to load the state dict, for which it doesn't matter what the task/extensions are -> all we need is the state dict of weights!

Reply:
Perhaps if you have an example of where we need to pass this argument when converting weights that would be most helpful!

Reply:
This is still not resolved ! Could you provide an example as suggested above ?